### PR TITLE
Add GitHub Workflow Replacement for Jenkins job_PreCommit_Portable_Python

### PR DESCRIPTION
**Pull request by aleksandr-dudko**
The following jenkins jobs have been migrated:

https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Portable_Python_Commit/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Portable_Python_Cron/
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Portable_Python_Phrase/

How has it been tested?
The job has been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

by creating pull requests with changes in trigger paths
by adding a comment to the PR
by making a push
Screenshot from 2023-06-29 17-10-43

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job_PreCommit_Portable_Python.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PreCommit Portable Python
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths: ['model/**', 'runners/core-construction-java/**', 'runners/core-java/**', 'runners/extensions-java/**', 'runners/flink/**', 'runners/java-fn-execution/**', 'runners/reference/**', 'sdks/python/**', 'release/**', '.github/workflows/job_PreCommit_Portable_Python.yml']
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_Portable_Python.yml:22]: Сould you list the paths on each line to make it more readable? Here is an example - https://github.com/apache/beam/blob/master/.github/workflows/beam_PreCommit_Java_Examples_Dataflow.yml#L25-L31

File: .github/workflows/job_PreCommit_Portable_Python.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            ${{matrix.python_version}}
+            3.8
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          cache-read-only: false
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: run Portable Python script
+        run: PY_VER=${{ matrix.python_version }} ; ./gradlew :sdks:python:test-suites:portable:py${PY_VER//.}:preCommitPy${PY_VER//.}
```
Comment (andreydevyatkin) [inline .github/workflows/job_PreCommit_Portable_Python.yml:95]: We need to add the default gradle switches (https://github.com/apache/beam/blob/master/.test-infra/jenkins/CommonJobProperties.groovy#L179) as the Jenkins job uses it: https://github.com/apache/beam/blob/master/.test-infra/jenkins/job_PreCommit_Portable_Python.groovy#L51

File: .github/workflows/job_PreCommit_Portable_Python.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  PortablePythonPreCommit:
```
Comment (magicgoody) [inline .github/workflows/job_PreCommit_Portable_Python.yml:72]: job name should start from beam_PreCommit_*

Comment (aleksandr-dudko) [reply .github/workflows/job_PreCommit_Portable_Python.yml]: It's done.

**Review by andreydevyatkin — APPROVED** [review]
LGTM

**Review by damccorm — APPROVED** [review]
LGTM

**Pull request merged**